bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -2415,55 +2481,77 @@ void BCStateTran::processData() {
     // if we have a new block
     //////////////////////////////////////////////////////////////////////////
     const uint64_t firstRequiredBlock = psd_->getFirstRequiredBlock();
+    bool lastBlock = isGettingBlocks && (firstRequiredBlock >= nextRequiredBlock_);
     if (!lastCollectedBlockId_) lastCollectedBlockId_ = firstRequiredBlock;
     if (newBlockIsValid && isGettingBlocks) {
-      DataStoreTransaction::Guard g(psd_->beginTransaction());
       sourceSelector_.setSourceSelectionTime(currTime);
 
       ConcordAssertAND(lastChunkInRequiredBlock >= 1, actualBlockSize > 0);
-      bool lastBlock = (firstRequiredBlock >= nextRequiredBlock_);
 
       // Report collecting status for every block collected. Log entry is created every fixed window
       // getMissingBlocksSummaryWindowSize If lastBlock is true: summarize the whole cycle without including "commit
       // to chain duration" and vblock. In that case last window might be less than the fixed
       // getMissingBlocksSummaryWindowSize
       reportCollectingStatus(firstRequiredBlock, actualBlockSize, lastBlock);
-      if (lastBlock) {
-        commitToChainDT_.start();
-        blocks_collected_.pause();
-        bytes_collected_.pause();
-      } else {
-        putBlocksStTempDT_.start();
-      }
-      betweenPutBlocksStTempDT_.pause();
-      LOG_DEBUG(getLogger(), "Add block: " << std::boolalpha << KVLOG(lastBlock, nextRequiredBlock_, actualBlockSize));
-      {
-        TimeRecorder scoped_timer(*histograms_.dst_put_block_duration);
-        ConcordAssert(as_->putBlock(nextRequiredBlock_, buffer_, actualBlockSize));
-      }
+
+      // Put the block. We distinguishe between last block non-last block
+      LOG_DEBUG(
+          getLogger(),
+          "Before putBlock id " << nextRequiredBlock_ << ":" << std::boolalpha << KVLOG(lastBlock, actualBlockSize));
       if (!lastBlock) {
         putBlocksStTempDT_.pause();
         betweenPutBlocksStTempDT_.start();
-        as_->getPrevDigestFromBlock(nextRequiredBlock_,
-                                    reinterpret_cast<StateTransferDigest *>(&digestOfNextRequiredBlock));
-        nextRequiredBlock_--;
-        g.txn()->setLastRequiredBlock(nextRequiredBlock_);
+        //////////////////////////////////////////////////////////////////////////
+        // Not the last block - block are committed into ST temporary blockchain
```

Comment:
ST doesn't know anything about temporary blockchain - it's a storage implementation details

Reply:
OK, I'll remove some part of the comment in 2 places.